CMakeLists.txt

```diff
@@ -230,8 +168,22 @@ add_executable(llvm2bpl
   tools/llvm2bpl/llvm2bpl.cpp
 )
 
+# We have to import LLVM's cmake definitions to build sea-dsa
+# Borrowed from sea-dsa's CMakeLists.txt
+find_package (LLVM 8.0.1 EXACT CONFIG NO_DEFAULT_PATH)
```

Comment:
Also, CMake ought to configure boost. Building on macOS with Boost from Homebrew fails like this:
```console
% (cd build && ninja)                                         
[2/57] Building CXX object CMakeFiles/smackTranslator.dir/lib/smack/Regions.cpp.o
FAILED: CMakeFiles/smackTranslator.dir/lib/smack/Regions.cpp.o 
/Library/Developer/CommandLineTools/usr/bin/c++  -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I../include -I../sea-dsa/include -I/usr/local/Cellar/llvm@8/8.0.1_1/include -std=c++11   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -fno-exceptions -fno-rtti -Wno-undefined-var-template -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -std=c++11 -w -fdiagnostics-color -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -MD -MT CMakeFiles/smackTranslator.dir/lib/smack/Regions.cpp.o -MF CMakeFiles/smackTranslator.dir/lib/smack/Regions.cpp.o.d -o CMakeFiles/smackTranslator.dir/lib/smack/Regions.cpp.o -c ../lib/smack/Regions.cpp
In file included from ../lib/smack/Regions.cpp:4:
In file included from ../include/smack/Regions.h:7:
../sea-dsa/include/sea_dsa/Graph.hh:3:10: fatal error: 'boost/container/flat_map.hpp' file not found
#include "boost/container/flat_map.hpp"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```
Adding the following fixes this:
```cmake
find_package (Boost 1.55)
if (Boost_FOUND)
  include_directories (${Boost_INCLUDE_DIRS})
endif ()
```
Shouldn’t this be present on all platforms?

I wonder how Linux builds are locating boost otherwise (?)

Reply:
It seems c++ compilers on Linux are able to locate the header files.

Reply:
Can you explain why? It seems a bit miraculous.